src/adapter/src/coord/sequencer.rs

```diff
@@ -3075,6 +3074,46 @@ impl<S: Append + 'static> Coordinator<S> {
             } else {
                 diffs
             };
+
+            // We need to clear out the read ops so the write doesn't fail in a
+            // read only transaction.
+            let ops = session.take_transaction_ops();
+            if let Some(ops) = ops {
```

Comment:
Can this ever be `None`? If not, should we assert that it's `Some`?

Reply:
This are some rare scenarios where this can be `None`. For example `INSERT INTO t SELECT now()` will result in this being `None`. See `transactions-stable.td` for some examples of that use case.